### DISCUSSION TITLE
Long start stream time after adding it from api with non persisten option

### DISCUSSION THREAD

**Discussion by krsvital** [OvenMediaLabs/OvenMediaEngine#1404] [Help]
After cheking logs i noticed, that server wait for 5 seconds to determine calculate method
Wait for RTCP SR for 5 seconds before starting the stream (stream.cpp:270)

then i commented all checks in function Stream::AjustRtpTimestamp and added string

_rtp_timestamp_method = RtpTimestampCalculationMethod::SINGLE_DELTA;

, recompile container and now stream starts after 2-3 seconds by webrtc vs 8-10 before

Can you add option to api named for example "defaultRtpCalculationMethod" which can be set method and disable useless waitings???

![image](https://github.com/AirenSoft/OvenMediaEngine/assets/14120204/bafb347e-cc16-4c0b-b4ed-bdd4c5105ec3)

**Comment by getroot** [comment]
Most WebRTC clients send RTCP immediately upon connection, so this process is almost instantaneous.
However, some RTSP servers do not send RTCP, so it is worth considering processing this as an option.
I will think about this for a moment.

**Reply by krsvital** [reply]
In my case RTSP server is RTSP camera and i tried many varios cameras and no one of them dont send RTCP

**Comment by krsvital** [comment]
What about adding new option to api named "defaultRtpCalculationMethod" for fixing this behavior in manual mode?

**Reply by getroot** [reply]
It's a great idea. However, I'm currently too busy to work on it. If you submit a pull request, I will review and merge it.

**Reply by krsvital** [reply]
#1418 pull request created
